Cubical/Homotopy/WhiteheadsLemma.agda

```diff
@@ -0,0 +1,447 @@
+                          ∙ λ i → (fst (r (~ i))) a)))
+
+  doubleCompPath-filler-refl : {A : Type ℓ} {a b : A} (p : a ≡ b)
+    → p ≡ refl ∙∙ p ∙∙ refl
+  doubleCompPath-filler-refl p = doubleCompPath-filler refl p refl
+
+  isContr→isContr→isEquiv : {A B : Type ℓ} (f : A → B)
+    → isContr A → isContr B → isEquiv f
+  equiv-proof (isContr→isContr→isEquiv f (a , hA) (b , hB)) b' =
+    ( a , isContr→isProp (b , hB) (f a) b') ,
+    ( λ (a' : fiber f b') → ΣPathP
+              ( isContr→isProp (a , hA) a (fst a')
+              , ( toPathP
+                  ( isProp→isSet
+                    ( isContr→isProp (b , hB))
+                    ( f _) b'
+                    ( transport (λ i → f _ ≡ b') _) (snd a')))))
+
+  isPointedTarget→isEquiv→isEquiv : {A B : Type ℓ} (f : A → B)
+    → (B → isEquiv f) → isEquiv f
+  equiv-proof (isPointedTarget→isEquiv→isEquiv f hf) =
+    λ y → equiv-proof (hf y) y
+
+  squareWithEquivs→Equiv : {A B C D : Type ℓ}
+    (f : A → B) (e1 : C → D) (e2 : A → C) (e3 : D → B)
+    → isEquiv e1 → isEquiv e2 → isEquiv e3 → e3 ∘ e1 ∘ e2 ≡ f
+    → isEquiv f
+  squareWithEquivs→Equiv f e1 e2 e3 e1Eq e2Eq e3Eq p =
+    transport
+    ( λ i → isEquiv (p i))
+    ( snd (compEquiv
+           ( e2 , e2Eq)
+           ( (e3 ∘ e1) , (snd (compEquiv (e1 , e1Eq) (e3 , e3Eq))))))
+-- end of the long private block of helper lemmas
+
+-- some useful paths between different maps
+
+{-
+
+a ≡ b ------> f a ≡ f b
+  |               ^
+  |               |
+  |               |
+  v               |
+a ≡ a ------> f a ≡ f a
+
+-}
+ΩCongSquare : {A B : Type ℓ} {a b : A} (f : A → B) (p : a ≡ b)
+  → (λ q → q ∙ (cong f p))
+   ∘ fst (Ω→ (f , refl))
+   ∘ (λ q → q ∙ sym p)
+   ≡ cong f
+ΩCongSquare f =
+  J ( λ b p → (λ q → q ∙ (cong f p))
+             ∘ fst (Ω→ (f , refl))
+             ∘ (λ q → q ∙ sym p)
+             ≡ cong f)
+    ( funExt (λ x → sym (rUnit _)
+                   ∙ sym (doubleCompPath-filler-refl _)
+                   ∙ cong (cong f) (sym (rUnit _))))
+
+private
+  setMapFunctorial : {A B C : Type ℓ} (f : A → B) (g : B → C)
+    → map g ∘ map f ≡ map (g ∘ f)
+  setMapFunctorial f g =
+    funExt (sElim (λ x → isSetPathImplicit) λ a → refl)
+
+  πHomπHomCongSquareAux : {A B : Type ℓ} {a : A} {n : ℕ} (f : A → B)
+    → Iso.inv (Iso-πΩ-π (1 + n))
+     ∘ (fst (πHom {A = (A , a)} {B = (B , f a)} (1 + n) (f , refl)))
+     ∘ (Iso.fun (Iso-πΩ-π (1 + n)))
+     ≡ map (Iso.inv (invIso (flipΩIso (1 + n)))
+         ∘ fst ((Ω^→ (2 + n)) (f , refl))
+         ∘ Iso.fun (invIso (flipΩIso (1 + n))))
+  πHomπHomCongSquareAux {n = n} f =
+      cong
+      ( λ g → g ∘ (Iso.fun (Iso-πΩ-π (1 + n))))
+      ( setMapFunctorial
+        ( fst ((Ω^→ (2 + n)) (f , refl)))
+        ( Iso.fun (flipΩIso (1 + n))))
+    ∙ setMapFunctorial
+      ( Iso.inv (flipΩIso (1 + n)))
+      ( Iso.fun (flipΩIso (1 + n)) ∘ fst ((Ω^→ (2 + n)) (f , refl)))
+
+-- Ω (Ω→ f) ≡ Ω→ (Ω f)
+PathPΩ^→Ω : {A B : Pointed ℓ} (n : ℕ) (f : A →∙ B)
```

Comment:
In the future we may consider moving some of these properties to Homotopy.Group.Base or something (or maybe a new Properties file?) but let's keep it here for now.